Update contributing docs and PR template according to our best practices (#779)
FerretDB/FerretDB: .github/PROCESS.md

2. Pull request **title** should be accurate and descriptive as it is used in the generated changelog.
   It should not end with a dot.
   It should not mention the issue number but describe the fixed issue instead.
3. Pull request **description** must mention the issue number like "Refs #123." or "Closes #456.".
   Do not use `https://github.com/org/repo/issue/123` even if it is rendered the same on GitHub.
   We provide a pull requests template that includes suggestions and readiness checklist.
   End sentences with a dot.
4. Commit messages (both titles and bodies) are not important and can be "WIP" or anything else.

>    All commits are always squashed on merge by GitHub, so there is no need to squash them manually, amend them, and/or do force pushes.

Reviewer: That line is about the development process and it's not so important as it's automated by squash commit setting.

>    Autogenerated GitHub's squash commit's title should be manually replaced by PR's title.

Reviewer: And this line is about the commit title and commit body which is important as we have to do it by ourselves. 
So I suggest extracting a separate number in this list.  WDYT of it?

Author: The need for this change is explained in the original issue by the author of the issue.

Reviewer: > The need for this change is explained in the original issue by the author of the issue.

So what do you think of extracting the separate number?.

>    Autogenerated GitHub's squash commit's body should be manually replaced by "Refs #{issue_number}". 

Reviewer 2: ```suggestion
   Autogenerated GitHub's squash commit's body should be manually replaced by "Refs #{issue_number}" or "Closes #{issue_number}". 
```

Reviewer: No here is a list with explanations of differences in between them:
* Closes closes issue automatically
* Refs mentions without closing the issue.

Reviewer 2: There is no mention of closing issues or whatever

Reviewer: > There is no mention of closing issues or whatever

But such questions will appear inevitably.
And when GitHub changes that behaviour we need to change it also.

   Do not leave original commit messages in the squashed commit body.

## Draft pull requests

> 1. We aim our issues to be scoped and well-described. 

Reviewer: Would you please rewrite in active voice?

Reviewer: Please see the comment https://github.com/FerretDB/FerretDB/pull/779#discussion_r904916865. 
I challenge the entire paragraph for its sense, existence, and meaning.

>    With that, ideally, there shouldn't be a need to create a draft pull request.

Reviewer: ```suggestion
   With that, ideally, no need to create a draft pull request.
```

Reviewer: > This suggestion is not grammatically correct (the verb is missing).

well, it's not clear then. 

The second sentence doesn't logically result from the first one. 

A draft is a draft and task decomposing is task decomposing. 

Why it should be connected somehow to our task decomposition?

2. We aim to implement and merge improvements as fast as possible to continuously increase 
   [Lead Time for Changes](https://cloud.google.com/blog/products/devops-sre/using-the-four-keys-to-measure-your-devops-performance). 

>    Having too many drafts indicates that something is wrong in our processes, it needs to be discussed and changed.

Reviewer: ```suggestion
   Too many drafts indicate that something is wrong in our processes. Then we need to discuss and change it.
```

3. Rarely, there is a need to demonstrate something that shouldn't be merged. 

>    For example, a corner case that reveals a bug in the code that can't be fixed easily.

Reviewer: Would you please rewrite in active voice?

```suggestion
   For example, a corner case reveals a bug in the code. And we cannot fix that bug easily.
```

>    For such cases, draft pull requests are possible.

Reviewer: It's not clear:
* if I create a draft PR with the test case for the bug? 
* If I create a PR with my functionality - so what is it for then?

Author: Yeah, let me remove this case, so the whole picture is less confusing.

4. If your changes need a review, create a normal pull request. 

>    After the first round of review, even if your changes are not accepted, continue working with a normal pull request.

Reviewer: Would you please add here `would you please`? :)

Reviewer: > Hmm... We don't really use "would you" in contributing/process docs. Does the sentence sound impolite without it?

yep, my impression of that sentence was, actually.

>    There is no need to switch between draft and non-draft. 

Reviewer: Looks like I have two PRs one is draft and another is not.
Would you please rewrite the sentence? 
Example: 
> You don't need to change the PR status to draft back and forth.

> 5. Please note that people don't get notifications about draft pull requests by default.

Reviewer: But code owners do - once a week or so?

Author: Well, it doesn't seem to be really significant. The point of that sentence is that if you want people to have a look at your draft, you should ask them and not assume that the team checks all the drafts. Maybe we should rephrase the whole sentence this way.
